extensions/classifiers/SVMPredictionService.js

```diff
@@ -42,8 +42,17 @@ oppia.factory('SVMPredictionService', ['$log', function($log) {
           }
           kvalues.push(Math.exp(-gamma * sum));
         }
-        return kvalues;
+      } else if (kernel == 'linear') {
```

Comment:
@seanlip, @anmolshkl and I think that something is wrong with this kernel. I will check its code with SVM's C++ code. This is pretty simple kernel so I don't think there should be any issues with this implementation.

Last time when I was working on tests for CodeReplPredictionService, I came across issues within native JS functions such as `sort()` which didn't sort the numeric list correctly and I had to pass a custom comparator so that it can be sorted. I think that we might have something similar issue here which has more to do with idiosyncrasies of JS then the code itself.